examples/client.rs

```diff
@@ -31,76 +33,86 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
         .with_span_events(tracing_subscriber::fmt::format::FmtSpan::FULL)
         .with_writer(std::io::stderr)
+        .with_max_level(tracing::Level::INFO)
         .init();
 
     let opt = Opt::from_args();
 
-    let dest = opt.uri.parse::<http::Uri>()?;
+    // DNS lookup
 
-    if dest.scheme() != Some(&http::uri::Scheme::HTTPS) {
-        Err("destination scheme must be 'https'")?;
+    let uri = opt.uri.parse::<http::Uri>()?;
+
+    if uri.scheme() != Some(&http::uri::Scheme::HTTPS) {
+        Err("uri scheme must be 'https'")?;
     }
 
-    let auth = dest
-        .authority()
-        .ok_or("destination must have a host")?
-        .clone();
+    let auth = uri.authority().ok_or("uri must have a host")?.clone();
 
     let port = auth.port_u16().unwrap_or(443);
 
-    // dns me!
     let addr = tokio::net::lookup_host((auth.host(), port))
         .await?
         .next()
         .ok_or("dns found no addresses")?;
 
-    eprintln!("DNS Lookup for {:?}: {:?}", dest, addr);
+    info!("DNS lookup for {:?}: {:?}", uri, addr);
 
-    // quinn setup
-    let tls_config_builder = rustls::ClientConfig::builder()
-        .with_safe_default_cipher_suites()
-        .with_safe_default_kx_groups()
-        .with_protocol_versions(&[&rustls::version::TLS13])?;
-    let mut tls_config = if !opt.insecure {
-        let mut roots = rustls::RootCertStore::empty();
-        match rustls_native_certs::load_native_certs() {
-            Ok(certs) => {
-                for cert in certs {
-                    if let Err(e) = roots.add(&rustls::Certificate(cert.0)) {
-                        eprintln!("failed to parse trust anchor: {}", e);
-                    }
+    // create quinn client endpoint
+
+    // load CA certificates stored in the system
+    let mut roots = rustls::RootCertStore::empty();
+    match rustls_native_certs::load_native_certs() {
+        Ok(certs) => {
+            for cert in certs {
+                if let Err(e) = roots.add(&rustls::Certificate(cert.0)) {
+                    error!("failed to parse trust anchor: {}", e);
                 }
             }
-            Err(e) => {
-                eprintln!("couldn't load any default trust roots: {}", e);
-            }
-        };
-        tls_config_builder
-            .with_root_certificates(roots)
-            .with_no_client_auth()
-    } else {
-        tls_config_builder
-            .with_custom_certificate_verifier(Arc::new(YesVerifier))
-            .with_no_client_auth()
+        }
+        Err(e) => {
+            error!("couldn't load any default trust roots: {}", e);
+        }
     };
+
+    // load certificate of CA who issues the server certificate
+    // NOTE that this should be used for dev only
+    if let Err(e) = roots.add(&rustls::Certificate(std::fs::read(opt.ca)?)) {
```

Comment:
Looks really good.
I\`m not sure. But when using the client to connect to other services, would trusting this Certificate enable some sort of MITM? 
Maybe it would be better to make the default of `ca` something that doesn\`t load a Certificate. This could also make the Client more trustworthy. What do you think?

Reply:
I happened to apply some mitm trick a while ago, to trick some web frontend on my device into displaying tampered results (prank on my friend). If this is what you're talking about, in order to pull it off, the attacker will at least need
* to be somewhat in control of the traffic so they can intercept the messages
* to somehow plant their CA certificate (not this one) in the victim